catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/index/IndexInitializer.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.spatial.geocoding.index;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutorService;
+
+import org.codice.ddf.spatial.geocoding.GeoEntryExtractionException;
+import org.codice.ddf.spatial.geocoding.GeoEntryExtractor;
+import org.codice.ddf.spatial.geocoding.GeoEntryIndexer;
+import org.codice.ddf.spatial.geocoding.GeoEntryIndexingException;
+import org.codice.ddf.spatial.geocoding.GeoNamesRemoteDownloadException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IndexInitializer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexInitializer.class);
+
+    private String defaultGeonamesDataPath;
+
+    private String indexLocationPath;
+
+    private GeoEntryExtractor extractor;
+
+    private GeoEntryIndexer indexer;
+
+    private ExecutorService executor;
+
+    public void init() {
+        File defaultGeonamesDataFile = Paths.get(defaultGeonamesDataPath)
+                .toFile();
+        File indexLocationDir = Paths.get(indexLocationPath)
+                .toFile();
```

Comment:
❓ I'm not quite familiar with how blueprint handles/communicates exceptions during an init method, but are we fine with the default behavior if `Paths.get`throws an `InvalidPathException`?

Reply:
If the distribution is good then this shouldn't really ever happen but I did run into this when developing and everything came up fine except for the this class.